task_processing/interfaces/event.py

```diff
@@ -1,13 +1,18 @@
 from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
 from pyrsistent import PRecord
 
-# TODO: organize and explain these
+from task_processing.interfaces.task_executor import TaskConfig
 
 
 class Event(PRecord):
     # reference to platform-specific event object
     raw = field()
     # is this the last event for a task?
     terminal = field(type=bool, mandatory=True)
+    success = field(type=(bool, type(None)), initial=None)
     # platform-specific event name
     platform_type = field(type=str)
+    task_config = field(type=TaskConfig)
```

Comment:
yeah, that's not ideal. task config at least isn't. it's sort of a tradeoff between having events passing more related stuff around or having to track a lot of data per executor "layer".

which bring me to executor "layers" - this (RetryingExecutor) is an attempt to take an aspect of execution framework that is orthogonal to actually running tasks and extract it into something that is composable. i've settled on executors to be composable, because they all implement the same interface: run/kill/stop, maybe it's wrong place to do it, but having composable layers is useful imo.

middleware_data is a place where different "middleware" (composable execution layers) can store some data useful for end-user. possibly bad naming. naming is hard.